src/Server.ts

```diff
@@ -1,31 +1,60 @@
-import * as express from 'express';
-class Server {
-// tslint:disable-next-line: semicolon
-app
-constructor(private config) {
-this.app = express();
-}
-bootstrap() {
-this.SetupRoutes();
-return this;
-}
-SetupRoutes() {
-const {app} = this;
-app.get('/health-check', (req, res, next) => {
-res.send('i am ok');
-});
-return this;
-}
-run() {
-const {app, config: {PORT}} = this;
-app.listen(PORT, (err) => {
-if (err) {
-console.log(err);
-}
-console.log(`App is running on port ${PORT}`);
-// tslint:disable-next-line: semicolon
-})
-}
-}
-export default Server;
+import * as express from "express";
+import * as bodyParser from "body-parser";
+import { notFoundHandler, errorHandler } from './libs/routes';
+
+class Server{
+    app
+    constructor(private config){
+        this.app=express()
+    }
+    bootstrap(){
+        this.setupRouts()
+        return this;
+    }
+    public setupRouts(){
+        const { app }=this;
+        app.use('/health-check',(req, res)=>{
+            console.log("inside Second middleware");
+            res.send("I am OK");
+        });
+
+        this.app.use(notFoundHandler);
+        this.app.use(errorHandler);
+        this.app.use((req, res, next) => {
+            next({
+                error: "Not Found",
+                code: 404
+                
+            })
+        })
 
+        this.app.use((err, req, res, next) => {
+            console.log(err);
+            res.json(
+                {
+                    "error ": err.error,
+                    status : err.code,
+                    message : err. message || "Error",
+                    timeStamp: new Date()
+                 
+                }
+            )
+        });
+        return this;
+    }
```

Comment:
Already created errorHandler and notFoundHandler, remove unnecessary code

Reply:
ok I will remove all unnecessary  code.